### PR TITLE
Rbs jtprbsfxirjtp 146 market service snapshot fucntionality

### PR DESCRIPTION
**Pull request by vasil-krastev**
Add SnapshotPersistenceTask that acts as a TimerTask
Add PersistScheduler to schedule a snapshot with JMX configurated frequency

### CODE REVIEW

File: market-generator/src/main/java/com/market/banica/generator/service/PersistSchedulerImpl.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+            } else {
+                LOGGER.warn("Please, provide a positive frequency!");
+            }
+        } else {
+            LOGGER.warn("The current frequency is the same!");
+        }
+    }
+
+    public void scheduleSnapshot() {
```
Comment (Lilyanov) [inline market-generator/src/main/java/com/market/banica/generator/service/PersistSchedulerImpl.java:60]: where is this method used ? The code between lines 46-50 can be replaced with this method

**Review by Lilyanov — APPROVED** [review]

File: market-generator/src/main/java/com/market/banica/generator/service/PersistSchedulerImpl.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.Timer;
+import java.util.concurrent.TimeUnit;
+
+@Component
+@ManagedResource
+public class PersistSchedulerImpl {
```
Comment (trayan-muchev) [inline market-generator/src/main/java/com/market/banica/generator/service/PersistSchedulerImpl.java:20]: Shouldn't this class implement the PersistScheduler interface?

Comment (vasilkrastev) [reply market-generator/src/main/java/com/market/banica/generator/service/PersistSchedulerImpl.java]: Fixed

**Review by trayan-muchev — APPROVED** [review]

File: market-generator/src/main/java/com/market/banica/generator/service/PersistSchedulerImpl.java (hunk 2 above)
Comment (vasil-krastev) [reply market-generator/src/main/java/com/market/banica/generator/service/PersistSchedulerImpl.java]: Fixed

**Pull request merged**